agents-common/src/main/java/org/apache/ranger/plugin/audit/RangerDefaultAuditHandler.java

```diff
@@ -41,288 +40,269 @@
+
+        LOG.debug("<== RangerDefaultAuditHandler.processResult({})", result);
+    }
+
+    @Override
+    public void processResults(Collection<RangerAccessResult> results) {
+        LOG.debug("==> RangerDefaultAuditHandler.processResults({})", results);
+
+        Collection<AuthzAuditEvent> events = getAuthzEvents(results);
+
+        if (events != null) {
+            logAuthzAudits(events);
+        }
+
+        LOG.debug("<== RangerDefaultAuditHandler.processResults({})", results);
+    }
+
+    public AuthzAuditEvent getAuthzEvents(RangerAccessResult result) {
+        LOG.debug("==> RangerDefaultAuditHandler.getAuthzEvents({})", result);
+
+        AuthzAuditEvent ret = null;
+
+        RangerAccessRequest request = result != null ? result.getAccessRequest() : null;
+
+        if (request != null && result.getIsAudited()) {
+            //RangerServiceDef     serviceDef   = result.getServiceDef();
+            RangerAccessResource resource     = request.getResource();
+            String               resourceType = resource == null ? null : resource.getLeafName();
+            String               resourcePath = resource == null ? null : resource.getAsString();
+
+            ret = createAuthzAuditEvent();
+
+            ret.setRepositoryName(result.getServiceName());
+            ret.setRepositoryType(result.getServiceType());
+            ret.setResourceType(resourceType);
+            ret.setResourcePath(resourcePath);
+            ret.setRequestData(request.getRequestData());
+            ret.setEventTime(request.getAccessTime() != null ? request.getAccessTime() : new Date());
+            ret.setUser(request.getUser());
+            ret.setAction(request.getAccessType());
+            ret.setAccessResult((short) (result.getIsAllowed() ? 1 : 0));
+            ret.setPolicyId(result.getPolicyId());
+            ret.setAccessType(request.getAction());
+            ret.setClientIP(request.getClientIPAddress());
+            ret.setClientType(request.getClientType());
+            ret.setSessionId(request.getSessionId());
+            ret.setAclEnforcer(moduleName);
+
+            Set<String> tags = getTags(request);
+            if (tags != null) {
+                ret.setTags(tags);
+            }
+
+            ret.setDatasets(getDatasets(request));
+            ret.setProjects(getProjects(request));
+            ret.setAdditionalInfo(getAdditionalInfo(request));
+            ret.setClusterName(request.getClusterName());
+            ret.setZoneName(result.getZoneName());
+            ret.setAgentHostname(restUtils.getAgentHostname());
+            ret.setPolicyVersion(result.getPolicyVersion());
+
+            populateDefaults(ret);
+
+            result.setAuditLogId(ret.getEventId());
+        }
+
+        LOG.debug("<== RangerDefaultAuditHandler.getAuthzEvents({}): {}", result, ret);
+
+        return ret;
+    }
+
+    public Collection<AuthzAuditEvent> getAuthzEvents(Collection<RangerAccessResult> results) {
```

Comment:
for readability:

```
public Collection<AuthzAuditEvent> getAuthzEvents(Collection<RangerAccessResult> results) {
    LOG.debug("==> RangerDefaultAuditHandler.getAuthzEvents({})", results);

    if (results == null || results.isEmpty()) {
        LOG.debug("<== RangerDefaultAuditHandler.getAuthzEvents({}): null", results);
        return null;
    }

    List<AuthzAuditEvent> ret = results.stream()
        .map(this::getAuthzEvents)
        .filter(Objects::nonNull)
        .collect(Collectors.toList());

    LOG.debug("<== RangerDefaultAuditHandler.getAuthzEvents({}): {}", results, ret);

    return ret.isEmpty() ? null : ret;
}
```

Reply:
As I responded earlier, the intention is to keep a single return from a method. I recommend not changing this approach.